### ISSUE TITLE
Provide support for restoring backed up data

### ISSUE THREAD

**Issue by george-hawkins-work** [rdegges/stormpath-export#1]
One of the stated purposes of stormpath-export in its README is:
- Back up your user data

Usually when you back up data you also want to be able to restore it, i.e. re-import it.

Currently the JSON produced by stormpath-export is essentially what you get from the _list_ calls like:

```
curl -u $API_KEY_ID:$API_KEY_SECRET \
     -H "Accept: application/json" \
     "https://api.stormpath.com/v1/tenants/$TENANT_ID/directories"
```

However it's not suitable for restoring data following an issue (e.g. you wrote some code that accidentally deleted all your users rather than just one). I.e. the JSON is **not** suitable for e.g. doing:

```
curl -X POST -u $API_KEY_ID:$API_KEY_SECRET \
    -H "Accept: application/json" \
    -H "Content-Type: application/json"
    --data @stormpath-exports/1yHawt280jqGyvHSbmGAXK/directories/1yozsKVqI1VWZPC81oGjxS.json
    https://api.stormpath.com/v1/directories
```

Unsurprisingly trying this results in `"developerMessage":"Property name 'createdAt' is invalid.  Property names may not equal any of the following reserved names: 'href', 'createdAt', 'modifiedAt', meta', 'spMeta', 'spmeta', 'ionmeta', or 'ionMeta'."` (actually you have to knock out the `"provider"` block before it'll start complaining about reserved names).

There are also circular dependencies between the various JSON files, e.g. JSON files for directories refer to groups while JSON files for groups refer to directories (and accounts).

So even if you knock out `createdAt` and other reserved names you can't restore your complete state, starting from a bare tenant, by rePOSTing these files one at a time.

In short you can export but being able to import would be nice too.

**Comment by rdegges** [comment]
You're 100% right.

The initial version I did of this was trying to just export the 'basic stuff', and I didn't consider that the objects would not be re-post'able to the API.

I had started building a separate import library, but then I stopped working on that as I changed my strategy a bit.

What I'd like to do is:
- Move this export functionality into the official CLI tool: https://github.com/stormpath/stormpath-cli
- Add an 'import' function to the CLI tool.
- Update the JSON blobs to more accurately reflect 'resource creation' -- this will simplify the import code.
- Add support for newer Stormpath resources.

So, that's essentially the gist of it atm. Thanks for this feedback -- super useful! I'll make sure this gets taken care of in the new CLI stuff ^^

**Comment by leoafarias** [comment]
Hey guys do we have an update on this?

**Comment by rdegges** [comment]
I have still not finished this. I've got some ideas about implementing this in the official CLI tool -- still need some time.
